vaadin-spring/pom.xml

```diff
@@ -67,7 +67,17 @@
             <artifactId>spring-boot-autoconfigure-processor</artifactId>
             <optional>true</optional>
         </dependency>
-
+        <dependency>
+            <groupId>org.springframework.data</groupId>
+            <artifactId>spring-data-commons</artifactId>
+            <optional>true</optional>
+        </dependency>
+        <dependency>
+            <groupId>com.vaadin</groupId>
+            <artifactId>flow-data</artifactId>
+            <optional>true</optional>
```

Comment:
Minor question: is that dependency have to be an optional indeed? It mean that the developer would have to add it explicitly in his project. Wouldn't it break a concept of using bill of materials (which has all of the Vaadin deps in one place). I don't have a strong experience with `optional`, thus just curious.

Reply:
Flow data is drawn in by all components that use it, so it can be optional here. No developer would use this method without having Vaadin component using flow-data in their project. In theory one could use flow without any components and then flow-data wouldn't be needed. In that case, people would need to exclude flow-data if they want to get rid of obsolete dependencies.

With the Spring Data dependency the optional is more relevant.

Sometimes when optional is used a lot, it is a sign of in-proper modularisation. We could have vaadin-spring-data module, where we'd have only this one method. But I'd not take this route yet for the sake of one helper method.

Reply:
Understood, thanks!